lib/Service/ViewService.php

```diff
@@ -449,13 +449,17 @@ private function enhanceView(View $view, string $userId): void {
 			$rawSortArray = $view->getSortArray();
 			if ($rawSortArray) {
 				$view->setSortArray(
-					array_map(static function ($sortRule) use ($view) {
+					array_map(static function ($sortRule) use ($view): array {
 						$columnsArray = $view->getColumnsArray();
-						if (isset($sortRule['columnId']) && $columnsArray && in_array($sortRule['columnId'], $columnsArray, true)) {
+						if (isset($sortRule['columnId'])
+							&& $sortRule['columnId'] >= 0 // meta columns can be exposed
```

Comment:
I would adapt to negated `isValidMetaTypeId()` check from https://github.com/nextcloud/tables/pull/1526 once that's in.